tools/crm_error.c

```diff
@@ -137,37 +225,45 @@ main(int argc, char **argv)
                 current++;
                 continue;
             }
-
-            if (options.with_name) {
-                // Room for four digits and a negative sign, space-padded
-                printf("% 5d: %-30s  %s\n", current, name, desc);
-            } else {
-                printf("% 5d: %s\n", current, desc);
-            }
+            out->message(out, "result-code", current,
+                         (options.with_name? name : NULL), desc);
             current++;
         }
 
     } else {
+        // For text output, print only "[name - ] description" by default
+        if (args->verbosity == 0) {
+            out->quiet = true;
+        }
+
         /* Skip #1 because that's the program name. */
         for (int lpc = 1; processed_args[lpc] != NULL; lpc++) {
+            int query = 0;
+
             if (pcmk__strcmp(processed_args[lpc], "--", pcmk__str_none) == 0) {
                 continue;
             }
-
-            pcmk__scan_min_int(processed_args[lpc], &rc, INT_MIN);
-            get_strings(rc, &name, &desc);
-            if (options.with_name) {
-                printf("%s - %s\n", name, desc);
-            } else {
-                printf("%s\n", desc);
-            }
+            pcmk__scan_min_int(processed_args[lpc], &query, INT_MIN);
+            get_strings(query, &name, &desc);
+            out->message(out, "result-code", query,
+                         (options.with_name? name : NULL), desc);
         }
     }
 
  done:
     g_strfreev(processed_args);
     pcmk__free_arg_context(context);
 
+    if (exit_code == CRM_EX_OK) {
+        exit_code = pcmk_rc2exitc(rc);
+    }
+
     pcmk__output_and_clear_error(error, NULL);
```

Comment:
You should pass `out` here as the second argument to pcmk__output_and_clear_error.

Reply:
Oops